ibm/mas_devops/roles/cp4d_wds/tasks/main.yml

```diff
@@ -0,0 +1,60 @@
+- name: "Fail if mas_instance_id is not provided"
+  when: mas_instance_id is not defined or mas_instance_id == ""
+  fail:
+    msg: "mas_instance_id property is required"
+
+- name: "Fail if mas_config_dir is not provided"
+  when: mas_config_dir is not defined or mas_config_dir == ""
+  fail:
+    msg: "mas_config_dir property is required"
+
+- name: "Fail if  is not provided"
+  when:
+    - assist_wds_pass is not defined or assist_wds_pass == ""
+    - useexternalwds is defined and useexternalwds
+  fail:
+    msg: "assist_wds_pass property is required"
+
+- name: "Fail if  is not provided"
+  when:
+    - assist_wds_admin is not defined or assist_wds_admin == ""
+    - useexternalwds is defined and useexternalwds
+  fail:
+    msg: "assist_wds_admin property is required"
+
+- name: "Fail if  is not provided"
+  when:
+    - assist_wds_url is not defined or assist_wds_url == ""
+    - useexternalwds is defined and useexternalwds
+  fail:
+    msg: "assist_wds_url property is required"
+
+
+- name: "cp4d/wds: Query CP4D Operator namespace"
+  when:
+    - useexternalwds is defined and useexternalwds
+  block:
+    - name: Get the CP4D instanceurl
+      set_fact:
+        wds_admin_username: "{{ assist_wds_admin }}"
+        wds_admin_password: "{{ assist_wds_pass }}"
+        wds_url: "{{ assist_wds_url  }}"
+    - name: Print the WDS related info
+      debug:
+        msg:
+          - "WDS url      .... {{ wds_url }}"
+          - "WDS Admin    .... {{ wds_admin_username }}"
+          - "WDS Passw0rd .... {{ wds_admin_password }}"
+
+- name: "cp4d/wds: Query CP4D Operator namespace"
+  when:
+    - not ( useexternalwds is defined and useexternalwds ) 
+  include_tasks: tasks/embeddedwds.yml
+  
+- name: Copy WDS secret to filesytem
```

Comment:
See earlier comment about making this conditional if mas_instance_id and mas_config_dir are set

Reply:
updated it